File: src/video/router.ts
import { Router } from 'express';

import {
  handleGetVideos,
  handleGetVideo,
  handleCreateVideo,
  handleUpdateVideo,
  handleDeleteVideo,
} from './controller';
import authMiddleware from '../auth/middleware';

const videoRouter = Router();

videoRouter.get('/', handleGetVideos);
videoRouter.get('/:id', handleGetVideo);
videoRouter.post('/', authMiddleware, handleCreateVideo);
videoRouter.put('/:id', authMiddleware, handleUpdateVideo);
videoRouter.delete('/:id', authMiddleware, handleDeleteVideo);

export default videoRouter;

File: videos/router.go
package videos

import (
	"github.com/AndreyGalchevski/strident-api/auth"
	"github.com/gorilla/mux"
)

func InitVideosRouter(r *mux.Router) {
	r.HandleFunc("/videos", handleGetVideos).Methods("GET")

	r.HandleFunc("/videos/{id}", auth.VerifyAuthorization(handleGetVideoByID)).Methods("GET")
	r.HandleFunc("/videos", auth.VerifyAuthorization(handlePostVideo)).Methods("POST")
	r.HandleFunc("/videos/{id}", auth.VerifyAuthorization(handlePatchVideo)).Methods("PATCH")
	r.HandleFunc("/videos/{id}", auth.VerifyAuthorization(handleDeleteVideo)).Methods("DELETE")
}
